overlord/registrystate/registrystate.go

```diff
@@ -199,3 +212,27 @@ func updateDatabags(st *state.State, databag registry.JSONDataBag, reg *registry
 	st.Set("registry-databags", databags)
 	return nil
 }
+
+type cachedRegistryTx struct{}
+
+// RegistryTransaction returns the registry.Transaction cached in the context
+// or creates one and caches it, if none existed. The context must be locked by
+// the caller.
+func RegistryTransaction(ctx *hookstate.Context, reg *registry.Registry) (*registry.Transaction, error) {
+	tx, ok := ctx.Cached(cachedRegistryTx{}).(*registry.Transaction)
```

Comment:
what happens if the hook access many registries at the same time? don't we need to key this also on registry account/name?

Reply:
Yes, that's a good point, I'll fix this and add a test